recipes/openssl/1.x.x/conanfile.py

```diff
@@ -241,17 +231,54 @@ def validate(self):
 
     def build_requirements(self):
         if self._settings_build.os == "Windows":
-            if not self._win_bash:
-                self.build_requires("strawberryperl/5.30.0.1")
-            if not self.options.no_asm and not tools.which("nasm"):
-                self.build_requires("nasm/2.15.05")
-        if self._win_bash and not tools.get_env("CONAN_BASH_PATH"):
+            if not self.win_bash:
+                self.tool_requires("strawberryperl/5.30.0.1")
+            if not self.options.no_asm:
+                self.tool_requires("nasm/2.15.05")
+        if self.win_bash and not os.getenv("CONAN_BASH_PATH") and not self._use_nmake:
             self.build_requires("msys2/cci.latest")
```

Comment:
I've not tested but I'm pretty sure it breaks MinGW. There is a confusion here about `self.win_bash`, it's not something set externally.

Reply:
I confirm, it's broken:

```
> conan install openssl/1.1.1s@ -o openssl:shared=True -pr:b win-x86_64-gcc12 -pr:h win-x86_64-gcc12 -b missing
Configuration (profile_host):
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.exception=seh
compiler.libcxx=libstdc++11
compiler.threads=posix
compiler.version=12
os=Windows
[options]
openssl:shared=True
[build_requires]
[env]
CC=D:/mingw64/bin/gcc.exe
CXX=D:/mingw64/bin/g++.exe
[conf]
tools.cmake.cmaketoolchain:generator=Ninja
tools.build:compiler_executables={'c': 'D:/mingw64/bin/gcc.exe', 'cpp': 'D:/mingw64/bin/gcc.exe'}
[buildenv]
CC=D:/mingw64/bin/gcc.exe
CXX=D:/mingw64/bin/g++.exe

Configuration (profile_build):
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.exception=seh
compiler.libcxx=libstdc++11
compiler.threads=posix
compiler.version=12
os=Windows
[options]
openssl:shared=True
[build_requires]
[env]
CC=D:/mingw64/bin/gcc.exe
CXX=D:/mingw64/bin/g++.exe
[conf]
tools.cmake.cmaketoolchain:generator=Ninja
tools.build:compiler_executables={'c': 'D:/mingw64/bin/gcc.exe', 'cpp': 'D:/mingw64/bin/gcc.exe'}
[buildenv]
CC=D:/mingw64/bin/gcc.exe
CXX=D:/mingw64/bin/g++.exe

openssl/1.1.1s: Not found in local cache, looking in remotes...
openssl/1.1.1s: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.18k]
Downloading conanfile.py completed [41.44k]
Downloading conan_export.tgz completed [0.33k]
Decompressing conan_export.tgz completed [0.00k]
openssl/1.1.1s: Downloaded recipe revision c5843f5e0776c9e862e08c911cd09958
strawberryperl/5.30.0.1: Not found in local cache, looking in remotes...
strawberryperl/5.30.0.1: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.10k]
Downloading conanfile.py completed [2.25k]
Downloading conan_export.tgz completed [0.31k]
Decompressing conan_export.tgz completed [0.00k]
strawberryperl/5.30.0.1: Downloaded recipe revision d125df083747d815c66e9ee621f3909f
nasm/2.15.05: Not found in local cache, looking in remotes...
nasm/2.15.05: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.22k]
Downloading conanfile.py completed [4.04k]
Downloading conan_export.tgz completed [0.31k]
Decompressing conan_export.tgz completed [0.00k]
nasm/2.15.05: Downloaded recipe revision c13d1db156147946c9aae400891b4baa
Installing package: openssl/1.1.1s
Requirements
    openssl/1.1.1s from 'conancenter' - Downloaded
Packages
    openssl/1.1.1s:bb11550f8236e219d08dd4d948c91d66592f7318 - Build
Build requirements
    nasm/2.15.05 from 'conancenter' - Downloaded
    strawberryperl/5.30.0.1 from 'conancenter' - Downloaded
Build requirements packages
    nasm/2.15.05:0a420ff5c47119e668867cdb51baff0eca1fdb68 - Download
    strawberryperl/5.30.0.1:ca33edce272a279b24f87dc0d4cf5bbdcffbc187 - Download

Installing (downloading, building) binaries...
nasm/2.15.05: Retrieving package 0a420ff5c47119e668867cdb51baff0eca1fdb68 from remote 'conancenter'
Downloading conanmanifest.txt completed [0.73k]
Downloading conaninfo.txt completed [0.32k]
Downloading conan_package.tgz completed [2044.13k]
Decompressing conan_package.tgz completed [0.00k]
nasm/2.15.05: Package installed 0a420ff5c47119e668867cdb51baff0eca1fdb68
nasm/2.15.05: Downloaded package revision 801ea6b6d3c17d507832dc2f9c17b6b8
strawberryperl/5.30.0.1: Retrieving package ca33edce272a279b24f87dc0d4cf5bbdcffbc187 from remote 'conancenter'
Downloading conanmanifest.txt completed [483.04k]
Downloading conaninfo.txt completed [0.30k]
Downloading conan_package.tgz completed [32402.67k]
Decompressing conan_package.tgz completed [0.00k]
strawberryperl/5.30.0.1: Package installed ca33edce272a279b24f87dc0d4cf5bbdcffbc187
strawberryperl/5.30.0.1: Downloaded package revision ffd124eb48d535d9e5b7134311ef0b65
openssl/1.1.1s: Applying build-requirement: strawberryperl/5.30.0.1
openssl/1.1.1s: Applying build-requirement: nasm/2.15.05
Downloading conan_sources.tgz completed [0.41k]
Decompressing conan_sources.tgz completed [0.00k]
[HOOK - conan-center.py] pre_source(): [IMMUTABLE SOURCES (KB-H010)] OK
openssl/1.1.1s: Configuring sources in C:\Users\spaceim\.conan\data\openssl\1.1.1s\_\_\source\src
Downloading openssl-1.1.1s.tar.gz completed [9637.68k]                                   openssl/1.1.1s: enssl/1.1.1s:
openssl/1.1.1s:
[HOOK - conan-center.py] post_source(): [LIBCXX MANAGEMENT (KB-H011)] OK
[HOOK - conan-center.py] post_source(): [CPPSTD MANAGEMENT (KB-H022)] OK
[HOOK - conan-center.py] post_source(): [SHORT_PATHS USAGE (KB-H066)] OK
openssl/1.1.1s: Copying sources to build folder
openssl/1.1.1s: Building your package in C:\Users\spaceim\.conan\data\openssl\1.1.1s\_\_\build\bb11550f8236e219d08dd4d948c91d66592f7318
openssl/1.1.1s: Generator txt created conanbuildinfo.txt
openssl/1.1.1s: Calling generate()
openssl/1.1.1s: Aggregating env generators
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] 'fPIC' option not found
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK
openssl/1.1.1s: Calling build()
openssl/1.1.1s: Apply patch (portability): TVOS and WatchOS don't like fork()
openssl/1.1.1s: gen_info = {'CFLAGS': ['-m64', '-O3'], 'CXXFLAGS': ['-m64', '-O3'], 'DEFINES': ['NDEBUG'], 'LDFLAGS': ['-m64']}
openssl/1.1.1s: using target: mingw-conan-Release-Windows-x86_64-gcc-12 -> mingw64
openssl/1.1.1s: my %targets = (
    "mingw-conan-Release-Windows-x86_64-gcc-12" => {
        inherit_from => [ "mingw64" ],
        cflags => add("-m64 -O3"),
        cxxflags => add("-m64 -O3"),
        defines => add(["NDEBUG"]),
        includes => add(),
        lflags => add("-m64"),



        cc => "D:/mingw64/bin/gcc.exe",
        cxx => "D:/mingw64/bin/g++.exe",



    },
);

openssl/1.1.1s: activated option: shared
openssl/1.1.1s: activated option: shared
openssl/1.1.1s: ['"mingw-conan-Release-Windows-x86_64-gcc-12"', 'shared', '--prefix=/', '--openssldir="res"', 'no-unit-test', 'threads', 'PERL=C:\\Users\\spaceim\\.conan\\data\\strawberryperl\\5.30.0.1\\_\\_\\package\\ca33edce272a279b24f87dc0d4cf5bbdcffbc187\\bin\\perl.exe', 'no-tests', '--release', 'no-md2', 'shared']
Configuring OpenSSL version 1.1.1s (0x1010113fL) for mingw-conan-Release-Windows-x86_64-gcc-12
Using os-specific seed configuration

******************************************************************************
This perl implementation doesn't produce Unix like paths (with forward slash
directory separators).  Please use an implementation that matches your
building platform.

This Perl version: 5.30.0 for MSWin32-x64-multi-thread
******************************************************************************
openssl/1.1.1s:
openssl/1.1.1s: ERROR: Package 'bb11550f8236e219d08dd4d948c91d66592f7318' build failed
openssl/1.1.1s: WARN: Build folder C:\Users\spaceim\.conan\data\openssl\1.1.1s\_\_\build\bb11550f8236e219d08dd4d948c91d66592f7318\build-release
ERROR: openssl/1.1.1s: Error in build() method, line 724
        self.run(f'{self._perl} ./Configure {args}')
        ConanException: Error 255 while executing C:\Users\spaceim\.conan\data\strawberryperl\5.30.0.1\_\_\package\ca33edce272a279b24f87dc0d4cf5bbdcffbc187\bin\perl.exe ./Configure "mingw-conan-Release-Windows-x86_64-gcc-12" shared --prefix=/ --openssldir="res" no-unit-test threads PERL=C:\Users\spaceim\.conan\data\strawberryperl\5.30.0.1\_\_\package\ca33edce272a279b24f87dc0d4cf5bbdcffbc187\bin\perl.exe no-tests --release no-md2 shared
```

Reply:
Hi @SpaceIm,
Sorry for breaking this. As you note, I had misunderstood the Conan 2.0 adoption of win_bash. Thanks for fixing it.